src/main/scala/com/github/shadowsocks/ProfileManagerActivity.scala

```diff
@@ -111,12 +102,9 @@ class ProfileManagerActivity extends AppCompatActivity with OnMenuItemClickListe
       }
     }
 
-    def onClick(v: View) = {
+    def onClick(v: View) {
       ShadowsocksApplication.switchProfile(item.id)
       finish
-      if (isLauncherFromShareIntent){
-        TaskStackBuilder.create(ProfileManagerActivity.this).addNextIntentWithParentStack(NavUtils.getParentActivityIntent(ProfileManagerActivity.this)).startActivities()
```

Comment:
If you want that, we can simply reverse #614. ProfileManagerActivity is a sub-activity of Shadowsocks. Don't break the hierarchy. If user want to switch profile, we now have quick switch.